pygmt/datatypes/dataset.py

```diff
@@ -226,11 +229,16 @@ def to_dataframe(
                 pd.Series(data=np.char.decode(textvector), dtype=pd.StringDtype())
             )
 
-        df = pd.concat(objs=vectors, axis="columns")
-        if column_names is not None:  # Assign column names
-            df.columns = column_names
-        if dtype is not None:
+        # Return an empty DataFrame if no columns are found.
+        if len(vectors) == 0:
```

Comment:
Have the inline comment match the indent of the `else:` block below.
```suggestion
        if len(vectors) == 0:
            # Return an empty DataFrame if no columns are found.
```